jupyter_server_fileid/manager.py

```diff
@@ -37,35 +37,174 @@ def wrapped(self, *args, **kwargs):
     return decorator
 
 
-class FileIdManager(LoggingConfigurable):
+class BaseFileIdManager(LoggingConfigurable):
```

Comment:
This should derive from `ABC` along with a metaclass definition in order for `@abstractmethod` decorators to be effective.  A good example of this can be found [here](https://github.com/jupyter/jupyter_client/blob/53ecd5b9cdb5fa5ac3403749664322524f15e6a4/jupyter_client/provisioning/provisioner_base.py#L20-L24).  By not deriving `BaseFileIdManager` from `ABC`, `@abstractmethod` decorators do not work (and I see those too have been removed). As a result, a subclass's violation for not implementing various methods will not be discovered until _that_ method is _called_, rather than when the class instance is instantiated.  Proper decoration also prevents `BaseFileIdManager` from being instantiated (i.e., a true abstract base class).